redfish-core/lib/hypervisor_system.hpp

```diff
@@ -527,25 +530,40 @@ inline void parseInterfaceData(nlohmann::json& jsonResponse,
     ipv4Array = nlohmann::json::array();
     ipv4StaticArray = nlohmann::json::array();
     bool ipv4IsActive = false;
+
+    constexpr int ipv4ConfigFieldsRequired = 4;
+    nlohmann::json::object_t ipv4;
     for (const auto& ipv4Config : ipv4Data)
     {
         if (ipv4Config.isActive)
         {
             ipv4IsActive = ipv4Config.isActive;
         }
-        if (!ipv4Config.address.empty())
+
+        if (!ipv4Config.origin.empty())
         {
-            nlohmann::json::object_t ipv4;
             ipv4["AddressOrigin"] = ipv4Config.origin;
+        }
+        if (!ipv4Config.netmask.empty())
+        {
             ipv4["SubnetMask"] = ipv4Config.netmask;
+        }
+        if (!ipv4Config.address.empty())
+        {
             ipv4["Address"] = ipv4Config.address;
+        }
+        if (!ipv4Config.gateway.empty())
+        {
             ipv4["Gateway"] = ipv4Config.gateway;
-
+        }
+        if (ipv4.size() == ipv4ConfigFieldsRequired)
+        {
             if (ipv4Config.origin == "Static")
             {
-                ipv4StaticArray.emplace_back(ipv4);
+                ipv4StaticArray.push_back(ipv4);
             }
             ipv4Array.emplace_back(std::move(ipv4));
+            ipv4 = nlohmann::json::object();
```

Comment:
L565 stmt should be moved to upper scope as each loop will need a new clean `ipv4`.
```
        if (ipv4.size() == 4)
        {
           ...
           ipv4Array.emplace_back(std::move(ipv4));
          ipv4 = nlohmann::json::object();
        }
      <-- Here the older info in ipv4 may split over to the next loop

==>

        if (ipv4.size() == 4)
        {
           ...
           ipv4Array.emplace_back(std::move(ipv4));
        }
        ipv4 = nlohmann::json::object();
```

In fact, `ipv4` is for each loop, and L534 is better moved to the inside the loop.

```
--- a/redfish-core/lib/hypervisor_system.hpp
+++ b/redfish-core/lib/hypervisor_system.hpp
@@ -531,9 +531,9 @@ inline void parseInterfaceData(nlohmann::json& jsonResponse,
     ipv4StaticArray = nlohmann::json::array();
     bool ipv4IsActive = false;

-    nlohmann::json::object_t ipv4;
     for (const auto& ipv4Config : ipv4Data)
     {
+        nlohmann::json::object_t ipv4;
         if (ipv4Config.isActive)
         {
             ipv4IsActive = ipv4Config.isActive;
@@ -562,7 +562,6 @@ inline void parseInterfaceData(nlohmann::json& jsonResponse,
                 ipv4StaticArray.push_back(ipv4);
             }
             ipv4Array.emplace_back(std::move(ipv4));
-            ipv4 =  nlohmann::json::object();
         }
     }
```

Reply:
Somehow the below problem didnot occur for ipv4. At one point in loop, I was able to see all the fields captured and then size becomes 4, and it will be pushed to ipv4Array.
I made the change here for ipv4 just to keep both ipv4 and ipv6 logics same.

Reply:
The current code may potentially spill ipv4Data to the next element if fields may be different between them.
For example,
 ipv4Data[0] has 3 fields on, and ipv4data[1] has 3 different fields on.

- [0] AddressOrigin1, SubnetMask1, Address1
- [1] AddressOrigin2, SubnetMask2,                 Gateway2

If so, the current code will create an entry with { AddressOrigin2, SubnetMask2,    Address1,   Gateway2}

Each loop must need to clear `ipv4`.  
This means that, `ipv4` is better to be defined at the top of each loop.

Reply:
Done. have made the change.